Update de.json for new terms and options

Tra1n84 opened the pull request:
### **Description**
- Enhanced the German translation file with new terms for better user experience.
- Added a description for maintaining Bluetooth connection.
- Updated the cutting mode terminology and introduced additional grid options.
- Included a new term for the cutting angle.


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>de.json</strong><dd><code>Update German translations with new terms and options</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

custom_components/mammotion/translations/de.json
<li>Added new translation terms for Bluetooth connection maintenance.<br> <li> Updated the cutting mode name and added states for grid options.<br> <li> Introduced a new term for cutting angle.<br>


</details>


  </td>
  <td><a href="https://github.com/mikey0000/Mammotion-HA/pull/113/files#diff-c28a07b60e03ec21803499d6c521fe20f2952f6a14901a5e279b77f54cc5b500">+12/-2</a>&nbsp; &nbsp; </td>
</tr>
</table></td></tr></tr></tbody></table>

mikey0000 commented:
@Tra1n84 could you please update the translation based on the branch I've just changed your PR to, Thanks! There are a few new fields and this is the next lot of work going out soon.

mikey0000 commented on custom_components/mammotion/translations/de.json:
```diff
@@ -168,6 +175,9 @@
         "name": "Aufgabengeschwindigkeit"
+      },
+      "cutting_angle": {
+        "name": "Wegwinkel (°)"
```
Please remove the angle symbol, this is provided by home assistant, I've updated the branch this PR is now pointing to for this entity to have the degrees symbol

Tra1n84 replied:
Okay. Good to know that 👍

mikey0000 commented on custom_components/mammotion/translations/de.json:
```diff
@@ -171,6 +177,9 @@
         "name": "Aufgabengeschwindigkeit"
+      },
+      "cutting_angle": {
+        "name": "Wegwinkel (°)"
```
```suggestion
        "name": "Wegwinkel"
```

Tra1n84 commented:
> @Tra1n84 could you please update the translation based on the branch I've just changed your PR to, Thanks! There are a few new fields and this is the next lot of work going out soon.

No problem. You're very welcome 😃 I have changed the de.json file and added german translations for new terms coming with that branch
